packages/slice-machine/src/modules/availableCustomTypes/types.ts

```diff
@@ -11,25 +11,29 @@ export type SyncedCustomType = {
 export type LocalFrontEndCustomType = NewCustomType | SyncedCustomType;
 
 export type DeletedFrontEndCustomType = {
-  local: undefined;
   remote: CustomTypeSM;
 };
 
 // TODO whats the difference between this and packages/slice-machine/lib/models/common/ModelStatus/compareCustomTypeModels.ts::FrontEndCtModel
 export type FrontEndCustomType =
-  | LocalFrontEndCustomType
+  | NewCustomType
+  | SyncedCustomType
   | DeletedFrontEndCustomType;
 
 export const isLocalCustomType = (
   ct: FrontEndCustomType
-): ct is LocalFrontEndCustomType => ct.local !== undefined;
+): ct is LocalFrontEndCustomType => "local" in ct && ct.local !== undefined;
 
 export const isDeletedCustomType = (
   ct: FrontEndCustomType
 ): ct is DeletedFrontEndCustomType => !isLocalCustomType(ct);
 
 export const isNewCustomType = (ct: FrontEndCustomType): ct is NewCustomType =>
-  isLocalCustomType(ct) && !("remote" in ct);
+  !("remote" in ct);
+
+export const isSyncedCustomType = (
+  ct: FrontEndCustomType
+): ct is SyncedCustomType => "remote" in ct && "local" in ct;
```

Comment:
❓ Is this enough? We don't need to check that local & remote !== undefined?

Reply:
remote being undefined is not a valid use case. It's either present and a custom type, or not defined at all

Reply:
See [here](https://github.com/prismicio/slice-machine/blob/964279b1e08823d215c38c0680569c9cd47e4afd/packages/slice-machine/src/modules/availableCustomTypes/types.ts#L4).
An attribute being undefined is what caused the crash we had before.